src/main/java/io/github/dediamondpro/xcatch/gui/ViewGui.java

```diff
@@ -136,17 +144,21 @@ public void onInventoryClick(InventoryClickEvent event) {
                     default:
                         if (event.getSlot() < 45 && event.getCurrentItem() != null && (event.getCurrentItem().getType() == Material.RED_CONCRETE
                                 || event.getCurrentItem().getType() == Material.YELLOW_CONCRETE)) {
-                            List<String> lore = event.getCurrentItem().getItemMeta().getLore();
-                            String[] coordinates = lore.get(1).replace("§7", "").split(" ");
-                            if (coordinates.length < 3) return;
+                            Integer actionIndex = event.getCurrentItem().getItemMeta().getPersistentDataContainer().get(
+                                    XCatch.INSTANCE.getActionDataKey(), PersistentDataType.INTEGER
+                            );
+                            if (actionIndex == null) return;
+                            ActionData actionData = PersistentData.data.actions.get(uuid).get(actionIndex);
                             try {
                                 HashMap<String, String> variables = new HashMap<String, String>() {{
                                     put("{player}", name);
-                                    put("{x}", coordinates[0]);
-                                    put("{y}", coordinates[1]);
-                                    put("{z}", coordinates[2]);
+                                    put("{world}", XCatch.INSTANCE.getServer().getWorld(actionData.worldUID).getName());
+                                    put("{x}", String.valueOf(actionData.x));
+                                    put("{y}", String.valueOf(actionData.y));
+                                    put("{z}", String.valueOf(actionData.z));
                                 }};
                                 event.getWhoClicked().closeInventory();
+                                XCatch.INSTANCE.getLogger().info(Utils.replaceVariables(XCatch.config.getString("view-click-command"), variables));
```

Comment:
Is this log needed?

Reply:
Nope, forgot to remove it when debugging world names, good catch.